core/src/main/java/tc/oc/pgm/score/ScoreMatchModule.java

```diff
@@ -72,21 +76,24 @@ public void load() {
   }
 
   public boolean hasScoreLimit() {
-    return this.config.scoreLimit > 0 || hasMercyRule();
+    return scoreLimit > 0 || hasMercyRule();
   }
 
   public boolean hasMercyRule() {
     return this.mercyRule != null;
   }
 
   public int getScoreLimit() {
-    checkState(hasScoreLimit());
+    // checkState(hasScoreLimit());
+    return scoreLimit;
+  }
 
-    if (hasMercyRule()) {
-      return this.mercyRule.getScoreLimit();
-    }
```

Comment:
You need to keep dynamic calls to mercy rule stuff, that changes dynamically thru the match, you cannot hard-code a value to return.

Just add a little `if (scoreLimitOverride > 0) return this.scoreLimitOverride;` before the mercy-rule check